File: Trabalho_Final/TrabalhoFinal/src/servidor/Esqueleto.java
package servidor;

import org.json.JSONObject;

public class Esqueleto {
    
	CalculadoraCiclista calculadora= new CalculadoraCiclista();
        
	public String IMC(JSONObject json) {
            double altura = json.getDouble("altura");
            double peso = json.getDouble("peso");
            return calculadora.IMC(altura, peso);
	}
	public String alturaBike(JSONObject json) {
            double altura = json.getDouble("altura");
            return calculadora.alturaBike(altura);
	}
	public String batimentosCardiacos(JSONObject json) {
            double batimentosTotal = json.getDouble("batimentosTotal");
            return calculadora.batimentosCardiacos(batimentosTotal);
	}
}

File: Trabalho_Final_SD/SistemasDistribuidos-master/Trabalho_Final/TrabalhoFinal/src/servidorJS/Esqueleto.js
const Services = require('./Services');

class Esqueleto {
    services = Services;

    IMC(argumentos) {
        const imc = new this.services.IMC(...argumentos);
        return imc.calcIMC();
    }
    Bike(argumentos) {
        const bike = new this.services.Bike(...argumentos);
        return bike.alturaBike();
    }
    Health(argumentos) {
        const health = new this.services.Health(...argumentos);
        return health.batimentosCardiacos();
    }
}

module.exports = Esqueleto;